### PR TITLE
10 get /api/articles queries

### PR DESCRIPTION
**Pull request by jaketysome**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -325,16 +325,99 @@ describe("api", () => {
...
+                votes: expect.any(Number),
+                comment_count: expect.any(Number),
+              })
+            );
+          });
+        });
+    });
+    test("status: 200, should return an empty array if there are no articles with the provided topic", () => {
+      return request(app)
+        .get("/api/articles?topic=worlddomination")
```
Comment (Pipe-Smoking-Rabbit) [inline __tests__/app.test.js]: The testing for this has two parts - a `200` when the topic is valid and in the database ( but has no articles )
- and a `404` if the topic does not exist in the database

File: models/models.articles.js [hunk 2]
```diff
@@ -11,13 +11,43 @@ exports.selectTopics = () => {
...
+    "comment_count",
+  ];
+
+  const validOrderQueries = ["asc", "desc"];
+
+  if (!validSortByQueries.includes(sort_by)) {
+    return Promise.reject({ status: 400, msg: "bad request" });
+  }
+
+  if (!validOrderQueries.includes(order)) {
```
Comment (Pipe-Smoking-Rabbit) [inline models/models.articles.js]: I'd suggest cutting this down by merging these into one `if` statement using an `||` or operator

File: models/models.articles.js [hunk 3]
```diff
@@ -11,13 +11,43 @@ exports.selectTopics = () => {
   });
 };
 
-exports.selectArticles = () => {
+exports.selectArticles = (topic, sort_by = "created_at", order = "desc") => {
+  const validSortByQueries = [
```
Comment (Pipe-Smoking-Rabbit) [inline models/models.articles.js:15]: very nice use of default value assignment

**Review by Pipe-Smoking-Rabbit — CHANGES_REQUESTED** [review]
Very smoothly done! Missing one test for `404` when the topic is not in the database ( I think you're currently testing that but with a `200` - but that's only if the topic is actually in the database ) Add that in and then feel free to merge

File: __tests__/app.test.js (hunk 1 above)
Comment (jaketysome) [reply __tests__/app.test.js]: Done

File: models/models.articles.js (hunk 2 above)
Comment (jaketysome) [reply models/models.articles.js]: Done!

File: models/models.articles.js (hunk 3 above)
Comment (jaketysome) [reply models/models.articles.js:15]: Cheers!

**Pull request merged**